src/Dialect/Krnl/KrnlHelper.cpp

```diff
@@ -158,6 +158,41 @@ void KrnlIterateOperandPack::pushAffineMapBound(
     _operands.emplace_back(operand);
 }
 
+// Bound could be a constant, Value or AffineMap
+void KrnlIterateOperandPack::pushIndexExprBound(IndexExpr expr) {
+  if (expr.isLiteral()) {
+    pushConstantBound(expr.getLiteral());
+  } else if (expr.isAffine() && !expr.isPredType()) {
+    int dimNum = expr.getContext().getNumDims();
```

Comment:
Since this functionality may eventually be used in other place, maybe add a IndexExpr::getAffineMap()

then you have a getAffineMap(), getDimAndSymbolList(), and pushAffineMapBound? that would be pretty clean interface.

Reply:
I tried to call this function in pushBounds(int64_t, IndexExpr), some test cases in check-onnx-backend-dynamic failed. There seems to be more work. I suggest we put all them into another patch.